lua/orgmode/org/hyperlinks.lua

```diff
@@ -45,32 +91,39 @@ function Hyperlinks.find_by_dedicated_target(base, skip_mapping)
   return targets
 end
 
-function Hyperlinks.find_by_title(base, skip_mapping)
-  if not base or base == '' then
+function Hyperlinks.find_by_title(ctx)
+  if not ctx.base or ctx.base == '' then
     return {}
   end
-  local headlines = Files.get_current_file():find_headlines_by_title(base)
-  if skip_mapping then
+  local headlines = Files.get_current_file():find_headlines_by_title(ctx.base)
+  if ctx.skip_add_prefix then
     return headlines
   end
   return vim.tbl_map(function(headline)
     return headline.title
   end, headlines)
 end
 
-function Hyperlinks.find_matching_links(base, skip_mapping)
-  base = vim.trim(base)
-  local prefix = base:sub(1, 1)
-  if prefix == '#' then
-    return Hyperlinks.find_by_custom_id_property(base, skip_mapping)
+function Hyperlinks.find_matching_links(ctx)
+  ctx = ctx or {}
+  ctx.base = ctx.base and vim.trim(ctx.base) or nil
+
+  update_hyperlink_ctx(ctx)
+
+  if ctx.base == 'file:' and not ctx.skip_add_prefix then
```

Comment:
Update `ctx.base == 'file:'` to `ctx.base:find('^file:')`, since base will now contain items even after `file:`. `find_by_filepath` will properly filter that base.